pkg/codegen/main.go

```diff
@@ -250,3 +323,166 @@ func replaceImports(dir string) error {
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// processInterfaceFile processes the specified file and adds import, new struct line,
+// and new function line to the specified blocks within the file.
+func processInterfaceFile(path string, info os.FileInfo, err error) error {
+	const importPath = `"github.com/rancher/shepherd/pkg/session"`
+	const newStructLine = "\tts                *session.Session"
+	const newFuncLine = "\t\tts:                ts,"
+
+	if !info.IsDir() && strings.HasSuffix(info.Name(), "interface.go") {
+		fset := token.NewFileSet()
+		err := addImport(fset, path, importPath)
+		if err != nil {
+			return err
+		}
+
+		err = appendNewlineToBlockInFile(path, "group", newStructLine)
+		if err != nil {
+			return err
+		}
+
+		err = appendNewlineToBlockInFile(path, "&group", newFuncLine)
+		if err != nil {
+			return err
+		}
+
+		err = appendNewlineToBlockInFile(path, "version", newStructLine)
+		if err != nil {
+			return err
+		}
+
+		err = appendNewlineToBlockInFile(path, "&version", newFuncLine)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// appendNewlineToBlockInFile takes a path to a file, a code block(struct,return) in a file to update
+// and the string to insert in a new line within the block
+func appendNewlineToBlockInFile(filePath, blockName, newLine string) error {
+	// Read the file contents
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("error reading file: %v", err)
+	}
+
+	// Convert content to string and split into lines
+	input := string(content)
+	lines := strings.Split(input, "\n")
+	blockStart := -1
+	blockEnd := -1
+	braceCount := 0
+
+	// Create a regex for matching function declaration
+	var re *regexp.Regexp
+	re = regexp.MustCompile(fmt.Sprintf(`(?m)(type\s+%s\s+struct|\s*return\s*%s)\s*{\s*`, blockName, blockName))
+
+	// Find the start and end of the specified function
+	for i, line := range lines {
+		if blockStart == -1 {
+			if re.MatchString(line) {
+				blockStart = i
+				braceCount = 1
+			}
+		} else {
+			braceCount += strings.Count(line, "{") - strings.Count(line, "}")
+			if braceCount == 0 {
+				blockEnd = i
+				break
+			}
+		}
+	}
+
+	// If the target is found, insert the new line before the closing brace
+	if blockStart != -1 && blockEnd != -1 {
+		// Find the last non-empty line before the closing brace
+		insertPos := blockEnd
+		for i := blockEnd - 1; i > blockStart; i-- {
+			if strings.TrimSpace(lines[i]) != "" {
+				insertPos = i + 1
+				break
+			}
+		}
+
+		// Insert the new line
+		lines = append(lines[:insertPos], append([]string{newLine}, lines[insertPos:]...)...)
+
+		// Join the lines back together
+		modifiedContent := strings.Join(lines, "\n")
+
+		// Write the modified content back to the file
+		err = os.WriteFile(filePath, []byte(modifiedContent), 0644)
+		if err != nil {
+			return fmt.Errorf("error writing to file: %v", err)
+		}
+		return nil
+	}
+
+	return nil
```

Comment:
Maybe I am reading this part wrong, but return nil will be the upper one always right? 
```suggestion
	}

	return nil
```

Should it rely on the return nil at the bottom? Would this cause a scope issue?

Reply:
Fixed